src/main/java/com/google/sps/servlets/MealServlet.java

```diff
@@ -97,9 +98,24 @@ private void getMealList(HttpServletRequest request, HttpServletResponse respons
         // Every object Meal adds if it contains at least one keyword in at least one field 
         // (type, title, description, ingredients).
         List<Meal> searchedMeal = new ArrayList<>();
+        HashMap<Integer, List<Meal>> sortedMeal = new HashMap<>();
         for (Meal meal : meals) {
-            if (isResultOfSearch(meal, params)) {
-                searchedMeal.add(meal);
+            int frequency = getFrequency(meal, params);
+            if (frequency > 0) {
+                List<Meal> list = new ArrayList<>();
+                if (sortedMeal.get(frequency) != null) {
+                    list = sortedMeal.get(frequency);
+                }
+                list.add(meal);
+                sortedMeal.put(frequency, list);
```

Comment:
We should change the order of things that are going on here.
That's because in Java `List list = mymap.get(key);` will get you not a copy of the list, but reference to that list - if you modify the list in any way (with `.add()` etc) then the original object gets updated - the one that also happens to still be in the map.
Therefore we can do the following:
1. Make sure that there's a list under that key with `.putIfAbsent(new ArrayList<>());`
2. `.get()` the list (or use the result returned by `.putIfAbsent`, although since it sometimes returns `null` it'd be complicated)
3. Add an element to that list.